dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -642,19 +649,21 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
 
   void ProcessImagesJpeg2k(MixedWorkspace &ws) {
 #if NVJPEG2K_ENABLED
-    nvjpeg2k_thread_.AddWork([this, &ws](int) {
-      auto &output = ws.OutputRef<GPUBackend>(0);
-      auto &input = ws.InputRef<CPUBackend>(0);
-      for (auto *sample : samples_jpeg2k_) {
-        assert(sample);
-        auto i = sample->sample_idx;
-        auto output_data = output.mutable_tensor<uint8_t>(i);
-        auto in = span<const uint8_t>(input[i].data<uint8_t>(), input[i].size());
-        ImageCache::ImageShape shape = output_shape_[i].to_static<3>();
-        DecodeJpeg2k(output_data, sample, in);
-        CacheStore(sample->file_name, output_data, shape, nvjpeg2k_cu_stream_);
-      }
-    });
+    if (nvjpeg2k_handle_) {
```

Comment:
Similarly, I'd rather add an early return. But that might be just a personal preference so feel free to ignore

Reply:
Done